py/selenium/webdriver/chrome/webdriver.py

```diff
@@ -74,8 +75,11 @@ def __init__(
             )
         else:
             keep_alive = True
+        if not options:
+            options = self.create_options()
         if not service:
             service = Service(executable_path, port, service_args, service_log_path)
+        service.path = DriverFinder.get_path(service, options)
```

Comment:
hmm, except in Python we assign a default if one isn't passed in, so we wouldn't know the difference. And oddly "executable_path" in Python is actually "driver name"

Reply:
Well, we need to figure out how to let a user pass in a specific location for a driver without overriding it. I think this might mean an `@abstractmethod` method in the superclass for getting path and calling methods in the the subclass when executing `start()`.

Reply:
If the user sets the path, we do not overwrite it unless it does not work. That is done inside `DriverFinder`. I'll change it so we use a method to set the value.

Reply:
I made the change anyway, checking if there is a path set before doing anything. This aligns it with the code in the other bindings.

Reply:
Oh, I was thinking the default value in the Service class constructor would break this, but we want to support both user provided and what is on PATH, which the previous implementation would address. Would be really nice to have unit tests for these... :-D Might need @symonk help?

Reply:
Oh, that is a good catch. I always run tests before pushing, and they passed because I had copied a ChromeDriver to my path to test the changes to JavaScript locally and I forgot to remove it.
I now reverted the changes and it works fine. 
I am not sure how to test that as a unit or small test, because we need to actually interact with a file on the PATH. I'd prefer to create a new workflow where we remove the drivers from the PATH.

Reply:
Yeah, it would need a unit test with some kind of mocking, which I'm not as familiar with in pytest

Reply:
And in a future version I would like for Selenium Manager to process PATH instead of the bindings. Would love to pull out all the logic that searches all the directories for things. In which case we will need to differentiate what is passed in from the default that locates on PATH. But we can do that later.

Reply:
Processing PATH is already tracked in a different issue.